nifi-nar-bundles/nifi-smb-bundle/nifi-smb-processors/src/test/java/org/apache/nifi/processors/smb/SambaTestContainers.java

```diff
@@ -0,0 +1,92 @@
+ */
+package org.apache.nifi.processors.smb;
+
+import static java.util.Arrays.fill;
+import static org.apache.nifi.processors.smb.ListSmb.SMB_CLIENT_PROVIDER_SERVICE;
+import static org.apache.nifi.services.smb.SmbjClientProviderService.DOMAIN;
+import static org.apache.nifi.services.smb.SmbjClientProviderService.HOSTNAME;
+import static org.apache.nifi.services.smb.SmbjClientProviderService.PASSWORD;
+import static org.apache.nifi.services.smb.SmbjClientProviderService.PORT;
+import static org.apache.nifi.services.smb.SmbjClientProviderService.SHARE;
+import static org.apache.nifi.services.smb.SmbjClientProviderService.USERNAME;
+
+import org.apache.nifi.services.smb.SmbjClientProviderService;
+import org.apache.nifi.util.TestRunner;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.builder.Transferable;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+public class SambaTestContainers {
+
+    protected final static Integer DEFAULT_SAMBA_PORT = 445;
+    protected final static Logger logger = LoggerFactory.getLogger(SambaTestContainers.class);
+    protected final GenericContainer<?> sambaContainer = new GenericContainer<>(DockerImageName.parse("dperson/samba"))
+            .withExposedPorts(DEFAULT_SAMBA_PORT, 139)
+            .waitingFor(Wait.forListeningPort())
+            .withLogConsumer(new Slf4jLogConsumer(logger))
+            .withCommand("-w domain -u username;password -s share;/folder;;no;no;username;;; -p");
+
+    @BeforeEach
+    public void beforeEach() {
+        sambaContainer.start();
+    }
+
+    @AfterEach
+    public void afterEach() {
+        sambaContainer.stop();
+    }
+
+    protected SmbjClientProviderService configureTestRunnerForSambaDockerContainer(TestRunner testRunner)
+            throws Exception {
+        SmbjClientProviderService smbjClientProviderService = new SmbjClientProviderService();
+        testRunner.addControllerService("client-provider", smbjClientProviderService);
+        testRunner.setProperty(SMB_CLIENT_PROVIDER_SERVICE, "client-provider");
+        testRunner.setProperty(smbjClientProviderService, HOSTNAME, sambaContainer.getHost());
+        testRunner.setProperty(smbjClientProviderService, PORT,
+                String.valueOf(sambaContainer.getMappedPort(DEFAULT_SAMBA_PORT)));
+        testRunner.setProperty(smbjClientProviderService, USERNAME, "username");
+        testRunner.setProperty(smbjClientProviderService, PASSWORD, "password");
+        testRunner.setProperty(smbjClientProviderService, SHARE, "share");
+        testRunner.setProperty(smbjClientProviderService, DOMAIN, "domain");
+        return smbjClientProviderService;
```

Comment:
I think it would make sense to enable the controller service here instead of all individual tests.
```suggestion
        testRunner.setProperty(smbjClientProviderService, DOMAIN, "domain");
        
        testRunner.enableControllerService(smbjClientProviderService);
        
        return smbjClientProviderService;
```